forgerock-auth/src/main/java/org/forgerock/android/auth/DefaultSingleSignOnManager.java

```diff
@@ -38,55 +41,66 @@ class DefaultSingleSignOnManager implements SingleSignOnManager, ResponseHandler
 
     private static final String TAG = DefaultSingleSignOnManager.class.getSimpleName();
 
-    private final SingleSignOnManager singleSignOnManager;
+    private final Storage<SSOToken> ssoTokenStorage;
+    private final Storage<Collection<String>> cookiesStorage;
     private final ServerConfig serverConfig;
     private final SSOBroadcastModel ssoBroadcastModel;
     private static final Action LOGOUT = new Action(Action.LOGOUT);
 
     @Builder
     private DefaultSingleSignOnManager(@NonNull Context context,
                                        ServerConfig serverConfig,
-                                       SSOBroadcastModel ssoBroadcastModel,
-                                       SharedPreferences sharedPreferences) {
-        Logger.warn(TAG, "Fallback to SharedPreference to store SSO Token");
-        singleSignOnManager = SharedPreferencesSignOnManager.builder()
-                .context(context)
-                .sharedPreferences(sharedPreferences).build();
+                                       Storage<SSOToken> ssoTokenStorage,
+                                       Storage<Collection<String>> cookiesStorage,
+                                       SSOBroadcastModel ssoBroadcastModel) {
+
+        this.ssoTokenStorage = ssoTokenStorage == null ? Options.INSTANCE.getSsoTokenStorage() : ssoTokenStorage;
```

Comment:
We don't use SharedPreferencesSignOnManager anymore, we switch to Storage interface to store tokens